ghc-src/Miso/Html/Internal.hs

```diff
@@ -114,7 +114,19 @@ instance L.ToHtml (VTree action) where
                      , "autocomplete"
                      ]
         toTag = T.toLower
-        kids = foldMap L.toHtml vChildren
+        kids
+          = foldMap L.toHtml
+          . V.fromList
+          . collapseSiblingTextNodes
+          . V.toList
+          $ vChildren
+
+collapseSiblingTextNodes :: [VTree a] -> [VTree a]
+collapseSiblingTextNodes [] = []
+collapseSiblingTextNodes (VText x : VText y : xs) =
+  collapseSiblingTextNodes (VText (x <> " " <> y) : xs)
```

Comment:
Should there really be a space in between? I would expect multiple text nodes to be concatenated directly.